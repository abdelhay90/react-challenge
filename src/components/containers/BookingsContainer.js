/**
 * Container to hold the bookings feature and assign the trip store to it
 */
import React from 'react';
import { inject, observer } from 'mobx-react';
import Bookings from '../bookings/Bookings';

const BookingsContainer = inject('trip')(
  observer(({ trip }) => {
    return <Bookings passengers={trip.passengers} />;
  }),
);

export default BookingsContainer;
