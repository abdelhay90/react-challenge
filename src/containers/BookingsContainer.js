import React from 'react';
import { inject, observer } from 'mobx-react';
import Bookings from '../components/bookings/Bookings';

const BookingsContainer = inject('trip')(
  observer(({ trip }) => {
    return <Bookings passengers={trip.passengers} />;
  }),
);

export default BookingsContainer;
