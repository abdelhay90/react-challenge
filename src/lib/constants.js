export const MAP_TOKEN =
  'pk.eyJ1IjoiYWhtZWQtYWJkZWxoYXkiLCJhIjoiY2swb2lkeGcyMGE1dDNlcnV3bzYyeXBqYSJ9.n0kF9SCChITXNQispwGnYQ';
export const FEATURE_COLLECTION_TYPE = 'FeatureCollection';
export const FEATURE_TYPE = 'Feature';
export const GEOMETRY_POINT_TYPE = 'Point';
export const GEOMETRY_LINE_TYPE = 'LineString';
export const LAYER_LINE_TYPE = 'line';
export const LAYER_CIRCLE_TYPE = 'circle';
export const LAYER_SYMBOL_TYPE = 'symbol';

export const DEFAULT_LAYER_CIRCLE_PAINT = {
  'circle-color': '#ff5200',
  'circle-stroke-width': 1,
  'circle-stroke-color': '#fff',
  'circle-stroke-opacity': 1,
};

export const DEFAULT_LAYER_LINE_PAINT = {
  'line-width': 2,
  'line-color': '#007cbf',
};

export const DEFAULT_LAYER_SYMBOL_BUS_PAINT = {
  'icon-image': 'airport-15',
  'icon-rotate': ['get', 'bearing'],
  'icon-rotation-alignment': 'map',
  'icon-allow-overlap': true,
  'icon-ignore-placement': true,
};

// 30.1 seconds, the .1 is to allow a buffer for the next set of cords to load
export const ANIMATION_STEPS = 30100;

export const TRIP_STATUS = {
  STARTED: 'STARTED',
  NOT_STARTED: 'NOT_STARTED',
  FINISHED: 'FINISHED',
  IN_PROGRESS: 'IN_PROGRESS',
};

export const PAYMENT_METHOD = {
  CASH: 'CASH',
  CREDIT: 'CREDIT',
};

export const PASSENGER_STATUS = {
  WAITING: 'WAITING',
  CHECKED_IN: 'CHECKED_IN',
  MISSED: 'MISSED',
  CANCELLED: 'CANCELLED',
  COMPLETED: 'COMPLETED',
};
