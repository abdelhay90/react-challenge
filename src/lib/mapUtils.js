import { along, lineDistance, lineString } from '@turf/turf';
import mapboxgl from 'mapbox-gl';
import {
  FEATURE_COLLECTION_TYPE,
  FEATURE_TYPE,
  GEOMETRY_POINT_TYPE,
} from './constants';

/**
 * fit map view port to contain and zoom to specific shape
 * @param map
 * @param shape
 */
export const fitBounds = (map, shape) => {
  const startBounds = new mapboxgl.LngLatBounds();
  startBounds.extend(shape[0]);
  const bounds = shape.reduce((newBounds, coord) => {
    return newBounds.extend(coord);
  }, startBounds);

  map.fitBounds(bounds, {
    padding: 20,
    duration: 0,
  });
};

/**
 * create a map feature and mutate new object that can be used in maps and feature collection
 * @param item
 * @param geometryType
 * @param coordinates
 * @returns {{geometry: {coordinates: *, type: *}, type: *, properties}}
 */
export const createFeature = (
  item,
  geometryType = GEOMETRY_POINT_TYPE,
  coordinates = [0, 0],
) => {
  return {
    type: FEATURE_TYPE,
    properties: { ...item },
    geometry: {
      type: geometryType,
      coordinates,
    },
  };
};

/**
 * create feature collection from array of data
 * @param items
 * @param geometryType
 * @returns {{features: *, type: *}}
 */
export const createFeatureCollection = (items, geometryType) => {
  return {
    type: FEATURE_COLLECTION_TYPE,
    features: items.map(item =>
      createFeature(item, geometryType, item.coordinates),
    ),
  };
};

/**
 * calculate an arc that make marker movement on map smooth
 * @param lineCoords
 * @param distanceOptions
 * @param steps
 * @returns {[]}
 */
export const calculateSmoothRouteArc = (
  lineCoords,
  { distanceOptions, steps },
) => {
  const arc = [];
  const line = lineString(lineCoords); // our array of lat/lngs
  const distance = lineDistance(line, distanceOptions);
  for (let i = 0; i < distance; i += distance / steps) {
    const segment = along(line, i, distanceOptions);
    arc.push(segment.geometry.coordinates);
  }

  return arc;
};

/**
 * adding new Geo JSON layer to the associated map
 * @param map
 * @param sourceOptions
 * @param layerOptions
 */
export const addGeoJSONLayer = ({ map, sourceOptions, layerOptions }) => {
  map.addSource(sourceOptions.name, {
    type: 'geojson',
    data: sourceOptions.data,
  });
  map.addLayer({ ...layerOptions });
};
