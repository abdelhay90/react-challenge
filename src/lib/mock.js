export const stops = [
  {
    stationId: '5d865ee507bfe196bac94ff3',
    stationName: 'Talaat Harb Axis',
    coordinates: [29.995097117000945, 31.445976804476345].reverse(),
  },
  {
    stationId: '5d865ee543724f1184914907',
    stationName: 'Omar Ibn Abdelaziz',
    coordinates: [30.012880226989733, 31.431022685763764].reverse(),
  },
  {
    stationId: '5d865ee54ec4f96a85fa3dbc',
    stationName: 'Southern Teseen',
    coordinates: [30.016423452010145, 31.43356512343098].reverse(),
  },
  {
    stationId: '5d865ee54fs346a85fa3dbc',
    stationName: 'AlMoshir Tantawy Axis',
    coordinates: [30.016375545253084, 31.39839364118893].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a85fa3dbc',
    stationName: 'El-Shaikh Kamel Khedr',
    coordinates: [30.040987895343214, 31.34613896298947].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a86trh6bc',
    stationName: 'Ahmed ElZomor',
    coordinates: [30.04505095423728, 31.34143552756268].reverse(),
  },
  {
    stationId: '432756654ec4f96a85fa3dbc',
    stationName: 'Mostafa ElNahas',
    coordinates: [30.054089469238654, 31.34217149725646].reverse(),
  },
  {
    stationId: '5d8756654ec4f96a85fa3987',
    stationName: 'Makram Ebid',
    coordinates: [30.0619822837204, 31.34523262262769].reverse(),
  },
  {
    stationId: '5d8734554ec4f96a85fa3dbc',
    stationName: 'Makrem Ebeid Ext',
    coordinates: [30.073225027534153, 31.343785066473743].reverse(),
  },
  {
    stationId: '5d87566598u4f96a85fa3dbc',
    stationName: 'ElThawra St.',
    coordinates: [30.08201257054215, 31.343882169980134].reverse(),
  },
];

export const passengers = [
  {
    id: '5d865ee507bfe196bac94ff3',
    name: 'Bailey Walls',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee543724f1184914907',
    name: 'Angie Rice',
    status: 'MISSED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee54ec4f96a85fa3dbc',
    name: 'Cecilia Trevino',
    status: 'CANCELLED',
    paymentMethod: 'credit',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5ad0a247f264dd75f',
    name: 'Coffey Fleming',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5ecfd4d136af665ba',
    name: 'Stuart Orr',
    status: 'CANCELLED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee5b3da2fb19e62a897',
    name: 'Vickie Beard',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee525b0567a26b1c2f3',
    name: 'Sandra Blackburn',
    status: 'COMPLETED',
    paymentMethod: 'CASH',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '432756654ec4f96a85fa3dbc',
  },
  {
    id: '5d865ee5f18d6587c1cb9ed0',
    name: 'Tia Oneal',
    status: 'CANCELLED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee527d6024df9300d92',
    name: 'Marylou Carr',
    status: 'MISSED',
    paymentMethod: 'CREDIT',
    imageUrl: 'static/images/full-face-2.png',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee51049173f6c386500',
    name: 'Rowe Rocha',
    status: 'MISSED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/leo.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d87566598u4f96a85fa3dbc',
  },
  {
    id: '5d865ee5d0d97ea49d4e5124',
    name: 'Mcconnell Jefferson',
    status: 'COMPLETED',
    paymentMethod: 'CREDIT',
    imageUrl: '/static/images/full-face-1.jpg',
    pickupStationId: '5d865ee507bfe196bac94ff3',
    checkoutStationId: '5d8734554ec4f96a85fa3dbc',
  },
  // {
  //   id: '5d865ee580c33a45ed14c2ed',
  //   name: 'Cook Best',
  //   status: 'CANCELLED',
  //   paymentMethod: 'CREDIT',
  //   imageUrl: '/static/images/full-face-1.jpg',
  //   pickupStationId: '5d865ee507bfe196bac94ff3',
  //   checkoutStationId: '5d87566598u4f96a85fa3dbc'
  // },
];

export const trip = {
  id: '123456789',
  distance: 47,
  rateFare: 25,
  driverInfo: {
    carModel: ' TOYOTA HiAce 2019',
    carImageUrl: '/static/images/Toyota-HiAce.jpg',
    driverName: 'Ahmed Abdelhay',
    driverImageUrl: '/static/images/driver-1.jpg',
    plateNumber: 'QA 2378',
  },
  path: stops.slice(0),
  passengers: passengers.slice(0),
};
