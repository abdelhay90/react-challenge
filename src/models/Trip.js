import { observable, action, computed } from 'mobx';
import groupBy from 'lodash/groupBy';
import Passenger from './Passenger';
import {
  PASSENGER_STATUS,
  PASSENGERS_MAX_CAPACITY,
  TRIP_STATUS,
} from '../lib/constants';

export default class Trip {
  id;

  @observable distance;

  @observable rateFare;

  @observable driverInfo;

  @observable path;

  @observable passengers = [];

  @observable tripStatus;

  @observable currentStop;

  constructor({
    id,
    passengers = [],
    distance,
    rateFare,
    driverInfo,
    path,
    tripStatus = TRIP_STATUS.NOT_STARTED,
  }) {
    this.id = id;
    this.passengers = passengers.map(item => new Passenger({ ...item }, this));
    this.distance = distance;
    this.rateFare = rateFare;
    this.driverInfo = driverInfo;
    this.path = path;
    this.currentStop = this.path.length > 0 ? this.path[0] : null;
    this.tripStatus = tripStatus;
  }

  @computed
  get route() {
    return {
      name:
        this.path.length !== 0
          ? `${this.path[0].stationName} - ${this.path[this.path.length - 1].stationName}`
          : 0,
      route: this.path.map(item => item.coordinates),
      routeStops: this.path,
      startPoint: this.path.length !== 0 ? this.path[0] : '',
      endPoint: this.path.length !== 0 ? this.path[this.path.length - 1] : '',
      currentStop: this.currentStop,
    };
  }

  @computed
  get isTripMaxCapacity() {
    return this.passengers.length >= PASSENGERS_MAX_CAPACITY;
  }

  @computed
  get passengersStats() {
    let allGroups = Object.keys(PASSENGER_STATUS).reduce(
      (accum, value) => ({ ...accum, [value]: [] }),
      {},
    );
    const calculatedGroups = groupBy(this.passengers, 'status');
    allGroups = { ...allGroups, ...calculatedGroups };
    return Object.keys(allGroups).map(key => ({
      x: key,
      y: allGroups[key].length,
    }));
  }

  getStationMap() {
    return this.path.reduce(
      (accum, value) => ({ ...accum, [value.stationId]: [] }),
      {},
    );
  }

  combineStationStats(allStations, calculated) {
    const newStatsMap = { ...allStations, ...calculated };
    return Object.keys(newStatsMap).map(key => ({
      x: this.path.filter(item => item.stationId === key)[0].stationName,
      y: newStatsMap[key].length,
    }));
  }

  @computed
  get stationPickupStats() {
    const allStations = this.getStationMap();
    const passengers = this.passengers.filter(
      passenger =>
        passenger.status !== PASSENGER_STATUS.MISSED &&
        passenger.status !== PASSENGER_STATUS.CANCELLED,
    );
    const stationGroups = groupBy(passengers, 'pickupStationId');
    return this.combineStationStats(allStations, stationGroups);
  }

  @computed
  get stationCheckoutStats() {
    const allStations = this.getStationMap();
    const passengers = this.passengers.filter(
      passenger => passenger.status === PASSENGER_STATUS.COMPLETED,
    );
    const stationGroups = groupBy(passengers, 'checkoutStationId');
    return this.combineStationStats(allStations, stationGroups);
  }

  @action.bound
  addPassenger(passenger) {
    if (this.passengers.length < 12)
      this.passengers.push(new Passenger({ ...passenger }, this));
  }

  @action.bound
  removePassenger(item) {
    this.passengers = this.passengers.filter(i => i !== item);
  }

  @action.bound
  startTrip() {
    this.tripStatus = TRIP_STATUS.STARTED;
  }

  @action.bound
  stopTrip() {
    this.tripStatus = TRIP_STATUS.FINISHED;
  }
}
