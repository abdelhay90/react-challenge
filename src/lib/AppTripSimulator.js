/**
 * create to simulate actual driver movement and his location update to update an use it on map
 */
import { EventEmitter } from 'events';
import { observe } from 'mobx';
import { calculateSmoothRouteArc } from './mapUtils';
import { ANIMATION_STEPS } from './constants';

class AppTripSimulator extends EventEmitter {
  constructor(trip) {
    super();
    this.trip = trip;
    this.init();
  }

  /**
   * init to observe the changes happened in the driver current location then calculate the
   * interpolated line between two point to make it easier while apply animation  on map
   */
  init() {
    observe(this.trip.driverInfo, change => {
      // to observe and detect changes to drivers location simulate smooth movement on map
      if (change.name === 'currentLocation') {
        const arc = calculateSmoothRouteArc(
          [change.oldValue, change.newValue],
          {
            distanceOptions: { units: 'feet' },
            steps: ANIMATION_STEPS,
          },
        );
        this.emit('location-change', arc);
      }
    });
  }

  /**
   * simulating driver push its location and setting current location
   */
  startSimulation() {
    const path = [...this.trip.path].reverse();
    // remove first location because it was bound when creating the trip path
    path.pop();

    const intervalId = setInterval(() => {
      const loc = path.pop();
      this.trip.driverInfo.currentLocation = loc.coordinates;
      if (path.length === 0) {
        clearInterval(intervalId);
      }
    }, 1000);
  }
}

export default AppTripSimulator;
