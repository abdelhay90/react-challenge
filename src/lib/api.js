import localforage from 'localforage';
import { trip, passengers } from './mock';

window.localforage = localforage;

const getAllTrips = async () => {
  const items = await localforage.getItem('trips');
  if (!items) await localforage.setItem('trips', [trip]);
  return items || [];
};

const getAllPassengers = async () => {
  const items = await localforage.getItem('passengers');
  if (!items) await localforage.setItem('passengers', [...passengers]);
  return items || [];
};

export default {
  async addTrip(item) {
    const items = await getAllTrips();
    const newItem = { ...item, id: Date.now() };
    await localforage.setItem('trips', [...items, newItem]);
    return newItem;
  },

  async getAllTrips() {
    const trips = await getAllTrips();
    return trips;
  },

  async deleteTrip({ id }) {
    const items = await getAllTrips();
    await localforage.setItem('trips', items.filter(item => item.id !== id));
  },

  async updateTrip(updatedItem) {
    const items = await getAllTrips();
    await localforage.setItem(
      'trips',
      items.map(item => {
        if (item.id === updatedItem.id) return { ...item, ...updatedItem };
        return item;
      }),
    );
  },

  async getAllPassengers() {
    const newPassengers = await getAllPassengers();
    return newPassengers;
  },
};
