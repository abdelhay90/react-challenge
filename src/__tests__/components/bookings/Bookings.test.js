import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import BookingsContainer from '../../../components/containers/BookingsContainer';
import Trip from '../../../models/Trip';
import { trip } from '../../../lib/mock';

describe('Bookings Component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    const currentTrip = new Trip({ ...trip });
    ReactDOM.render(
      <Provider trip={currentTrip}>
        <BookingsContainer />
      </Provider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should match snapshot', () => {
    const currentTrip = new Trip({ ...trip });
    const tree = renderer
      .create(
        <Provider trip={currentTrip}>
          <BookingsContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
