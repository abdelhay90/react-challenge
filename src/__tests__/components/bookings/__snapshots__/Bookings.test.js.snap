// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookings Component should match snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-153"
  >
    <div
      className="MuiBox-root MuiBox-root-154"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Bookings
      </h5>
    </div>
    <div
      className="MuiBox-root MuiBox-root-185"
    >
      <button
        aria-describedby={null}
        aria-label="Add Passenger"
        className="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-primary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add Passenger"
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-231"
  >
    <div
      className="MuiCollapse-container"
      style={
        Object {
          "minHeight": "300px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            className="MuiBox-root MuiBox-root-237"
          >
            <div
              className="MuiBox-root MuiBox-root-240 makeStyles-passenger-238"
              id="5d865ee507bfe196bac94ff3"
            >
              <div
                className="MuiBox-root MuiBox-root-241"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Bailey Walls"
                    className="MuiAvatar-img"
                    src="/static/images/leo.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-245"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Bailey Walls
                </h6>
                COMPLETED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-246 makeStyles-passenger-238"
              id="5d865ee543724f1184914907"
            >
              <div
                className="MuiBox-root MuiBox-root-247"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Angie Rice"
                    className="MuiAvatar-img"
                    src="/static/images/full-face-1.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-248"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Angie Rice
                </h6>
                MISSED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-249 makeStyles-passenger-238"
              id="5d865ee54ec4f96a85fa3dbc"
            >
              <div
                className="MuiBox-root MuiBox-root-250"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Cecilia Trevino"
                    className="MuiAvatar-img"
                    src="static/images/full-face-2.png"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-251"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Cecilia Trevino
                </h6>
                CANCELLED - credit
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-252 makeStyles-passenger-238"
              id="5d865ee5ad0a247f264dd75f"
            >
              <div
                className="MuiBox-root MuiBox-root-253"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Coffey Fleming"
                    className="MuiAvatar-img"
                    src="static/images/full-face-2.png"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-254"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Coffey Fleming
                </h6>
                COMPLETED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-255 makeStyles-passenger-238"
              id="5d865ee5ecfd4d136af665ba"
            >
              <div
                className="MuiBox-root MuiBox-root-256"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Stuart Orr"
                    className="MuiAvatar-img"
                    src="/static/images/leo.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-257"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Stuart Orr
                </h6>
                CANCELLED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-258 makeStyles-passenger-238"
              id="5d865ee5b3da2fb19e62a897"
            >
              <div
                className="MuiBox-root MuiBox-root-259"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Vickie Beard"
                    className="MuiAvatar-img"
                    src="/static/images/full-face-1.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-260"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Vickie Beard
                </h6>
                COMPLETED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-261 makeStyles-passenger-238"
              id="5d865ee525b0567a26b1c2f3"
            >
              <div
                className="MuiBox-root MuiBox-root-262"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Sandra Blackburn"
                    className="MuiAvatar-img"
                    src="/static/images/leo.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-263"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Sandra Blackburn
                </h6>
                COMPLETED - CASH
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-264 makeStyles-passenger-238"
              id="5d865ee5f18d6587c1cb9ed0"
            >
              <div
                className="MuiBox-root MuiBox-root-265"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Tia Oneal"
                    className="MuiAvatar-img"
                    src="/static/images/leo.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-266"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Tia Oneal
                </h6>
                CANCELLED - CREDIT
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-267 makeStyles-passenger-238"
              id="5d865ee527d6024df9300d92"
            >
              <div
                className="MuiBox-root MuiBox-root-268"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Marylou Carr"
                    className="MuiAvatar-img"
                    src="static/images/full-face-2.png"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-269"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Marylou Carr
                </h6>
                MISSED - CREDIT
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-270 makeStyles-passenger-238"
              id="5d865ee51049173f6c386500"
            >
              <div
                className="MuiBox-root MuiBox-root-271"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Rowe Rocha"
                    className="MuiAvatar-img"
                    src="/static/images/leo.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-272"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Rowe Rocha
                </h6>
                MISSED - CREDIT
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-273 makeStyles-passenger-238"
              id="5d865ee5d0d97ea49d4e5124"
            >
              <div
                className="MuiBox-root MuiBox-root-274"
              >
                <div
                  className="MuiAvatar-root makeStyles-bigAvatar-239"
                >
                  <img
                    alt="Mcconnell Jefferson"
                    className="MuiAvatar-img"
                    src="/static/images/full-face-1.jpg"
                  />
                </div>
              </div>
              <div
                className="MuiBox-root MuiBox-root-275"
                xs={6}
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Mcconnell Jefferson
                </h6>
                COMPLETED - CREDIT
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "left": "50%",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        View More
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
